crates/forge_repo/src/provider.rs

```diff
@@ -371,17 +395,40 @@ impl<F: EnvironmentInfra + FileReaderInfra + FileWriterInfra> ForgeProviderRepos
         configs.0
     }
 
+    /// Reads credentials from the JSON file with multiple fallback strategies:
+    /// 1. Try normal JSON parsing with `VecSkipError` to skip corrupt entries
+    /// 2. If JSON is syntactically broken, try to repair it with
+    ///    `forge_json_repair`
+    /// 3. If all else fails, return an empty Vec
     async fn read_credentials(&self) -> Vec<AuthCredential> {
         let path = self
             .infra
             .get_environment()
             .base_path
             .join(".credentials.json");
 
-        match self.infra.read_utf8(&path).await {
-            Ok(content) => serde_json::from_str(&content).unwrap_or_default(),
-            Err(_) => Vec::new(),
-        }
+        let Ok(content) = self.infra.read_utf8(&path).await else {
+            return Vec::new();
+        };
+
+        // Strategy 1: Try normal parsing with VecSkipError
+        serde_json::from_str::<ResilientCredentials>(&content)
+            .or_else(|_| {
+                // Strategy 2: Try JSON repair for syntactically broken JSON
+                tracing::warn!(path = %path.display(), "Failed to parse credentials file, attempting repair...");
+                forge_json_repair::json_repair::<ResilientCredentials>(&content).inspect(|_| {
+                    tracing::info!(path = %path.display(), "Successfully repaired credentials file");
+                })
+            })
+            .inspect_err(|e| {
+                tracing::error!(
+                    path = %path.display(),
+                    error = %e,
+                    "Failed to repair credentials file. Using empty credentials."
+                );
+            })
+            .map(|r| r.0)
+            .unwrap_or_default()
```

Comment:
This seems like a strategy we can use in multiple places. Can we abstract that out?